tidy3d/components/tcad/doping.py

```diff
@@ -70,11 +70,30 @@ def _get_indices_in_box(self, coords: dict, meshgrid: bool = True):
 
         return indices_in_box, X, Y, Z, normal_axis
 
+    @pd.root_validator(skip_on_failure=True)
+    def check_dimensions(cls, values):
+        """Make sure dimensionality is specified correctly. I.e.,
+        a 2D box must be defined with an inf size in the normal direction."""
+
+        size = values["size"]
+        for dim in range(3):
+            if size[dim] == 0:
+                dim_name = "xyz"[dim]
+                raise SetupError(
+                    f"The doping box has been setup with 0 size in the {dim_name} direction. "
+                    "If this was intended to be a 2D box, the box must have inf size in the "
```

Comment:
2D still sounds a bit undefined to me, whether it means infinitely extended or infinitely thin. How about something like "if this was intended to be translationally invariant, the box must have ..."